plugins/gfm/src/main/kotlin/org/jetbrains/dokka/gfm/renderer/CommonmarkRenderer.kt

```diff
@@ -306,6 +306,22 @@ open class CommonmarkRenderer(
         }
     }
 
+    override fun StringBuilder.buildCodeBlock(code: ContentCodeBlock, pageContext: ContentPage) {
+        append("```")
+        append(code.language.ifEmpty { "kotlin" })
+        buildNewLine()
+        code.children.forEach { it.build(this, pageContext) }
+        buildNewLine()
+        append("```")
+        buildNewLine()
```

Comment:
I wonder if this is necessary. Is there no newline or does it lead to some rendering bugs if we remove this?

Reply:
It depends on whether we can count on a newline being added by some higher level code, which I am not familiar enough with Dokka to predict. It is definitely safer to have extraneous newlines than missing newlines. As an example here:
```
Here is a block of literal text between triple backquotes.
``` And here text on the same line as the closing backquotes.
Here is text on the following line.

When the triple backquotes are not followed immediately by a newline, the remainder of the text is rendered as teletype.